h5py/h5o.pyx

```diff
@@ -84,13 +84,55 @@ cdef class _OHdr(_ObjInfoBase):
     property nmesgs:
         def __get__(self):
             return self.istr[0].hdr.nmesgs
+    property nchunks:
+        def __get__(self):
+            return self.istr[0].hdr.nchunks
+    property flags:
+        def __get__(self):
+            return self.istr[0].hdr.flags
 
     def __init__(self):
         self.space = _OHdrSpace()
         self.mesg = _OHdrMesg()
 
     def _hash(self):
-        return hash((self.version, self.nmesgs, self.space, self.mesg))
+        return hash((self.version, self.nmesgs, self.nchunks, self.flags, self.space, self.mesg))
+
+cdef class _MetaSizeObj(_ObjInfoBase):
+
+    property index_size:
+        def __get__(self):
+            return self.istr[0].meta_size.obj.index_size
+    property heap_size:
+        def __get__(self):
+            return self.istr[0].meta_size.obj.heap_size
+
+    def _hash(self):
+        return hash((self.index_size, self.heap_size))
+
+cdef class _MetaSizeAttr(_ObjInfoBase):
+
+    property index_size:
+        def __get__(self):
+            return self.istr[0].meta_size.attr.index_size
+    property heap_size:
+        def __get__(self):
+            return self.istr[0].meta_size.attr.heap_size
+
+    def _hash(self):
+        return hash((self.index_size, self.heap_size))
```

Comment:
I think these two classes can be condensed into one - we'll need two instances, but the corresponding C structs are of the same type (`H5_ih_info_t`).

Reply:
Yes, I tried to make them one but was struggling with the Cython code to access `attr` vs `obj` field from `self.istr[0].meta_size`.

Its all very new to me. I'll take another crack.

Reply:
I believe I got it working. It built and I was able to access the structs in ipython.

```
In [1]: import h5py

In [2]: h5py.__file__
Out[2]: '/home/delen/miniconda3/envs/h5py_dev/lib/python3.10/site-packages/h5py/__init__.py'

In [3]: from h5py.h5o import get_info

In [4]: h5 = h5py.File("./h5py/h5py/tests/data_files/vlen_string_dset.h5")

In [5]: get_info(h5['DS1'].id).meta_size.obj.index_size
Out[5]: 0

In [6]: get_info(h5['DS1'].id).meta_size.attr.index_size
Out[6]: 0

In [7]: get_info(h5['DS1'].id).meta_size.attr.heap_size
Out[7]: 0

In [8]: get_info(h5['DS1'].id).meta_size.obj.heap_size
Out[8]: 0
```